### PR TITLE
doc: Add documentation for special configurations

### PR DESCRIPTION
**Pull request by rakons**
This commit adds documentation for specialized configurations.

### CODE REVIEW

File: applications/machine_learning/README.rst [hunk 1]
```diff
@@ -242,6 +242,8 @@ The application supports the following build types:
 The given board can also support some additional configurations of the nRF Machine Learning application.
 For example, the nRF52840 Development Kit supports ``ZDebugNUS`` configuration that uses :ref:`nus_service_readme` instead of :ref:`zephyr:uart_api` for data forwarding.
 
+Any such special configiration would be described at the top of :file:`app_[Build].conf` file.
```
Comment (MarekPieta) [inline applications/machine_learning/README.rst:245]: configuration

Comment (MarekPieta) [reply applications/machine_learning/README.rst:245]: `app_[build_type].conf`?

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -395,6 +395,8 @@ For example, if the ``ZDebugWithShell`` build type is not supported on the selec
 In addition to the build types mentioned above, some boards can provide more build types, which can be used to generate an application in a specific variant.
 For example, such additional configurations are used to allow generation of application with different role (such as mouse, keyboard, or dongle on a DK) or to select a different link layer (such as LLPM capable Nordic SoftDevice LL or standard Zephyr SW LL).
 
+Such specific build variants are documented inside :file:`app_[Build].conf` file.
```
Comment (MarekPieta) [inline applications/nrf_desktop/README.rst:398]: `app_[build_type].conf?`

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf [hunk 3]
```diff
@@ -4,6 +4,9 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# MCUBootQSPI version
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf:7]: MCUBootSMP version

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf [hunk 4]
```diff
@@ -4,6 +4,9 @@
...
+# Configuration that enables MCUmgr SMP over BLE protocol
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf:8]: with DFU support

Comment (MarekPieta) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf:8]: We enable image management commands for that purpose

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf [hunk 5]
```diff
@@ -4,6 +4,9 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# MCUBootQSPI version
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf:7]: There are also configurations: `ZDebug_dongle`, `ZDebug_keyboard`

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug3BLEconn.conf [hunk 6]
```diff
@@ -4,6 +4,9 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# ZDebug3BLEconn version
+# The configuration that allows up to 3 simultaneous BLE connections.
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug3BLEconn.conf:8]: We could mention that LLPM support is disabled here (by default it's enabled)

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug4LLPMconn.conf [hunk 7]
```diff
@@ -4,6 +4,10 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# ZDebug3BLEconn version
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug4LLPMconn.conf:7]: ZDebug4LLPMconn

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZRelease4LLPMconn.conf [hunk 8]
```diff
@@ -4,6 +4,10 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# ZRelease3BLEconn version
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZRelease4LLPMconn.conf:7]: ZRelease4LLPMconn

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/app_ZDebugSplitLL.conf [hunk 9]
```diff
@@ -4,6 +4,9 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 ################################################################################
+# DebugSplitLL version
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/app_ZDebugSplitLL.conf:7]: I wonder if we should add the SplitLL configurations to the common list in README. They are used for quite a lot of boards.

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/machine_learning/README.rst (hunk 1 above)
Comment (rakons) [reply applications/machine_learning/README.rst:245]: Ok

File: applications/nrf_desktop/README.rst (hunk 2 above)
Comment (rakons) [reply applications/nrf_desktop/README.rst:398]: Ok

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf (hunk 3 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf:7]: Ok

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf (hunk 4 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootSMP.conf:8]: ok

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf (hunk 5 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/app_ZDebugMCUBootQSPI.conf:7]: ok

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug3BLEconn.conf (hunk 6 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug3BLEconn.conf:8]: ok

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug4LLPMconn.conf (hunk 7 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZDebug4LLPMconn.conf:7]: ok

File: applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZRelease4LLPMconn.conf (hunk 8 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840dongle_nrf52840/app_ZRelease4LLPMconn.conf:7]: ok

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/app_ZDebugSplitLL.conf (hunk 9 above)
Comment (rakons) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/app_ZDebugSplitLL.conf]: Done

**Review by MarekPieta — APPROVED** [review]

**Review by greg-fer — CHANGES_REQUESTED** [review]
Pushed a commit with changes and added some comments to .conf files. Please check and apply them as needed (and squash my edit).

**Comment by greg-fer** [comment]
Also, we need an entry to changelog :)

**Comment by greg-fer** [comment]
@kkasperczyk-no , @LuDuda , I see the weather station also includes a section about build types. Can we add a similar update there as the one proposed by Radoslaw in this PR?

**Review by pdunaj — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
> @kkasperczyk-no , @LuDuda , I see the weather station also includes a section about build types. Can we add a similar update there as the one proposed by Radoslaw in this PR?

As discussed offline, we probably don't plan to add another builds for now, so I believe we can perform alignment if necessary in the future.

File: applications/nrf_desktop/README.rst [hunk 10]
```diff
@@ -379,9 +379,11 @@ The following build types are available for various boards in the nRF Desktop:
 * ``ZRelease`` -- Release version of the application with no debugging features.
 * ``ZReleaseB0`` -- ``ZRelease`` build type with the support for the B0 bootloader enabled (for :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
 * ``ZReleaseMCUBoot`` -- ``ZRelease`` build type with the support for the MCUboot bootloader enabled (for :ref:`serial recovery DFU <nrf_desktop_bootloader_serial_dfu>` or :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
+* ``ZReleaseSplitLL`` -- ``ZRelease`` build type that selects Zephyr Bluetooth Link Layer.
 * ``ZDebug`` -- Debug version of the application; the same as the ``ZRelease`` build type, but with debug options enabled.
 * ``ZDebugB0`` -- ``ZDebug`` build type with the support for the B0 bootloader enabled (for :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
 * ``ZDebugMCUBoot`` -- ``ZDebug`` build type with the support for the MCUboot bootloader enabled (for :ref:`serial recovery DFU <nrf_desktop_bootloader_serial_dfu>` or :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`).
+* ``ZDebugSplitLL`` -- ``ZDebug`` build type that selects Zephyr Bluetooth Link Layer.
```
[the remark this answers is not in the document]
Comment (rakons) [reply applications/nrf_desktop/README.rst]: This is hard to say. Technically we should link to: https://docs.zephyrproject.org/latest/guides/bluetooth/bluetooth-arch.html?highlight=split#split . But the documentation is incomplete. We could possibly link to bluetooth-arch.

**Comment by rakons** [comment]
@greg-fer 
> Also, we need an entry to changelog :)

Do we? Are we documenting also such subtle documentation changes? No API nor functional changes are provided here.

**Review by greg-fer — APPROVED** [review]

**Pull request merged**